--- /Users/gnzlbg/py_nonblocking_orig.cpp	2014-07-31 11:07:26.000000000 +0200
+++ libs/mpi/src/python/py_nonblocking.cpp	2014-07-31 11:06:50.000000000 +0200
@@ -163,10 +163,11 @@
   {
     check_request_list_not_empty(requests);
     if (py_callable != object())
-      return test_all(requests.begin(), requests.end(), 
-          status_value_iterator(py_callable, requests.begin()));
+      return static_cast<bool>(
+          test_all(requests.begin(), requests.end(),
+                   status_value_iterator(py_callable, requests.begin())));
     else
-      return test_all(requests.begin(), requests.end());
+      return static_cast<bool>(test_all(requests.begin(), requests.end()));
   }
 
 
