Description: Change name of shared library
Author: Anton Gladky <gladky.anton@gmail.com>
Last-Update: 2012-11-28

--- a/wrappers/java/CMakeLists.txt
+++ b/wrappers/java/CMakeLists.txt
@@ -34,7 +34,11 @@
   set_target_properties(WrapGmsh PROPERTIES PREFIX "")
 endif(WIN32)
 if(UNIX)
-  set_target_properties(WrapGmsh PROPERTIES PREFIX "lib")
+  set_target_properties(WrapGmsh PROPERTIES 
+    PREFIX "lib"
+    OUTPUT_NAME java-gmsh
+    SOVERSION "${GMSH_MAJOR_VERSION}"
+  )
 endif(UNIX)		
 add_custom_command(TARGET WrapGmsh 
 POST_BUILD # do the rest of the command after the build period
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1048,7 +1048,7 @@
 # static library target
 if(ENABLE_BUILD_LIB)
   add_library(lib STATIC ${GMSH_SRC})
-  set_target_properties(lib PROPERTIES OUTPUT_NAME Gmsh)
+  set_target_properties(lib PROPERTIES OUTPUT_NAME gmsh)
   if(MSVC)
     set_target_properties(lib PROPERTIES DEBUG_POSTFIX d) 
     if(ENABLE_MSVC_STATIC_RUNTIME)
@@ -1060,10 +1060,10 @@
 # shared library target
 if(ENABLE_BUILD_SHARED OR ENABLE_WRAP_PYTHON OR ENABLE_WRAP_JAVA)
   add_library(shared SHARED ${GMSH_SRC})
-  set_target_properties(shared PROPERTIES OUTPUT_NAME Gmsh)
+  set_target_properties(shared PROPERTIES OUTPUT_NAME gmsh)
   set_target_properties(shared PROPERTIES 
                         VERSION ${GMSH_MAJOR_VERSION}.${GMSH_MINOR_VERSION}.${GMSH_PATCH_VERSION}
-                        SOVERSION ${GMSH_MAJOR_VERSION}.${GMSH_MINOR_VERSION})
+                        SOVERSION ${GMSH_MAJOR_VERSION})
   if(HAVE_LAPACK AND LAPACK_FLAGS)
     set_target_properties(shared PROPERTIES LINK_FLAGS ${LAPACK_FLAGS})
   endif(HAVE_LAPACK AND LAPACK_FLAGS)
