Description: Fix import of gmshpy
Author: Anton Gladky <gladk@debian.org>
Bug-Debian: http://bugs.debian.org/696420
Last-Update: 2013-01-26

--- a/wrappers/gmshpy/__init__.py.in
+++ b/wrappers/gmshpy/__init__.py.in
@@ -1,4 +1,7 @@
 # set Ctrl-C to default signal (terminates immediately)
+from ctypes import *
+mpi = CDLL('libmpi.so.0', RTLD_GLOBAL)
+
 import signal
 signal.signal(signal.SIGINT, signal.SIG_DFL)
 ${GMSH_PYTHON_MODULES_INCLUDE_CODE}
