Index: gmsh-tetgen-2.12.0/CMakeLists.txt
===================================================================
--- gmsh-tetgen-2.12.0.orig/CMakeLists.txt
+++ gmsh-tetgen-2.12.0/CMakeLists.txt
@@ -710,13 +710,15 @@ endif(ENABLE_MATHEX)
 if(ENABLE_MPI)
   find_package(MPI)
   if(MPI_FOUND)
+    SET(CMAKE_REQUIRED_INCLUDES "${MPI_INCLUDE_PATH};${CMAKE_REQUIRED_INCLUDES}")
+    INCLUDE_DIRECTORIES(${MPI_INCLUDE_PATH})
     set_config_option(HAVE_MPI "MPI")
     list(APPEND EXTERNAL_INCLUDES ${MPI_INCLUDE_DIR})
     list(APPEND EXTERNAL_LIBRARIES ${MPI_LIBRARIES})
-    include(CMakeForceCompiler)
-    # Warning: this actually requires cmake >= 2.8.5
-    cmake_force_c_compiler(${MPI_C_COMPILER} "MPI C Compiler")
-    cmake_force_cxx_compiler(${MPI_CXX_COMPILER} "MPI C++ Compiler")
+    # include(CMakeForceCompiler)
+    # # Warning: this actually requires cmake >= 2.8.5
+    # cmake_force_c_compiler(${MPI_C_COMPILER} "MPI C Compiler")
+    # cmake_force_cxx_compiler(${MPI_CXX_COMPILER} "MPI C++ Compiler")
   endif(MPI_FOUND)
 endif(ENABLE_MPI)
 
